In [ ]:
%lsmagics
%endpoint http://zoomathia.i3s.unice.fr/sparql

In [2]:
%show all
# Request whatever format is appropriate for the query type
%format default

# Activate table output
%display table

In [3]:
%prefix rdf:     <http://www.w3.org/1999/02/22-rdf-syntax-ns#> 
%prefix rdfs:    <http://www.w3.org/2000/01/rdf-schema#> 
%prefix oa:     <http://www.w3.org/ns/oa#>
%prefix skos: <http://www.w3.org/2004/02/skos/core#>
%prefix schema:  <http://schema.org/>
%prefix paragraph: <http://www.zoomathia.com/>

# CQ1 - Quels sont les animaux qui construisent un habitat ?

Cette question exprime le besoin du chercheur à cibler les passages dans le texte où l'auteur mentionne un animal et la construction d'un habitat.

   <u>__Reformulation__</u> : La reformulation de la requête SPARQL exprime cette requête comme étant la recherche: (1) d'une annotation Annotation ayant pour body le concept d'un animal et pour cible le paragraphe où cette mention est faite, (2) d'une Annotation ayant pour body le concept de construction d'habitat (enfant du concept "house building") et pour cible le paragraphe où cette mention est faite.

   <u>__Restriction__</u> : Pour restreindre les résultats et apporté de la cohérence entre les deux concepts, l'apparition des deux concepts mentionnés plus haut doivent faire partie du même paragraphe.

   <u>__Note__</u> : Deux collections rassemblent les concepts d'animaux: (1) la collection Zoonyme (Zoonyms) et (2) la collection Archéotaxon (Ancient class). Ces deux collections partagent une majeure partie des concepts animal. Les deux collections sont inclues pour maximiser le nombre de concept trouvé.


In [4]:
SELECT DISTINCT ?chapter_id ?paragraph_id ?text ?label_animal ?mention_habitat ?label_habitat 
WHERE {
     ?animal_annotation a oa:Annotation; oa:hasBody ?animal_concept;
        oa:hasTarget [
          oa:hasSource ?paragraph;
          oa:hasSelector [ oa:exact ?mention_animal;]].
    ?habitat_annotation a oa:Annotation; oa:hasBody ?habitat_concept;
        oa:hasTarget [
          oa:hasSource ?paragraph;
          oa:hasSelector [ oa:exact ?mention_habitat; ]].

    ?animal_concept a skos:Concept; skos:prefLabel ?label_animal.
    ?animal_collection a skos:Collection; skos:member ?animal_concept;
        skos:prefLabel ?label.
    
    ?habitat_concept a skos:Concept; skos:prefLabel ?label_habitat;
        skos:broader ?habitat_generic_concept.
    ?habitat_generic_concept a skos:Concept; skos:prefLabel "house building"@en.
    
    ?paragraph schema:text ?text;
               schema:identifier ?paragraph_id;
               schema:isPartOf ?chapter.
    
    ?chapter schema:identifier ?chapter_id.
    
    filter(lang(?label_animal) = "en")
    filter(lang(?label_habitat) = "en")
    filter(?label in ("Ancient class"@en, "Zoonyms"@en))
}
ORDER BY ?paragraph

chapter_id 
 paragraph_id 
 text 
 label_animal 
 mention_habitat 
 label_habitat 10 
 10 
 haliaetus tantum inplumes etiamnum pullos suos percutiens subinde cogit adversos intueri solis radios et, si coniventem umectantemque animadvertit, praecipitat e nido velut adulterinum atque degenerem. illum, cuius acies firma contra stetit, educat. 
 osprey 
 haliaetus tantum inplumes etiamnum pullos suos percutiens subinde cogit adversos intueri solis radios et, si coniventem umectantemque animadvertit, praecipitat e nido velut adulterinum atque degenerem. illum, cuius acies firma contra stetit, educat. 
 nest 10 
 10 
 haliaetus tantum inplumes etiamnum pullos suos percutiens subinde cogit adversos intueri solis radios et, si coniventem umectantemque animadvertit, praecipitat e nido velut adulterinum atque degenerem. illum, cuius acies firma contra stetit, educat. 
 eagle 
 haliaetus tantum inplumes etiamnum pullos suos percutiens subinde cogit adversos intueri solis radios et, si coniventem umectantemque animadvertit, praecipitat e nido velut adulterinum atque degenerem. illum, cuius acies firma contra stetit, educat. 
 nest 10 
 100 
 Perdices spina et frutice sic muniunt receptaculum, ut contra feram abunde vallentur. ovis stragulum molle pulvere contumulant nec in quo loco peperere incubant: ne cui frequentior conversatio suspecta sit, transferunt alio. illae quidem et maritos suos fallunt, quoniam intemperantia libidinis frangunt earum ova, ne incubando detineantur. tunc inter se dimicant mares desiderio feminarum; victum aiunt venerem pati. 
 WILD BEAST 
 Perdices spina et frutice sic muniunt receptaculum, ut contra feram abunde vallentur. ovis stragulum molle pulvere contumulant nec in quo loco peperere incubant: ne cui frequentior conversatio suspecta sit, transferunt alio. illae quidem et maritos suos fallunt, quoniam intemperantia libidinis frangunt earum ova, ne incubando detineantur. tunc inter se dimicant mares desiderio feminarum; victum aiunt venerem pati. 
 nest 10 
 100 
 Perdices spina et frutice sic muniunt receptaculum, ut contra feram abunde vallentur. ovis stragulum molle pulvere contumulant nec in quo loco peperere incubant: ne cui frequentior conversatio suspecta sit, transferunt alio. illae quidem et maritos suos fallunt, quoniam intemperantia libidinis frangunt earum ova, ne incubando detineantur. tunc inter se dimicant mares desiderio feminarum; victum aiunt venerem pati. 
 partridge 
 Perdices spina et frutice sic muniunt receptaculum, ut contra feram abunde vallentur. ovis stragulum molle pulvere contumulant nec in quo loco peperere incubant: ne cui frequentior conversatio suspecta sit, transferunt alio. illae quidem et maritos suos fallunt, quoniam intemperantia libidinis frangunt earum ova, ne incubando detineantur. tunc inter se dimicant mares desiderio feminarum; victum aiunt venerem pati. 
 nest 10 
 103 
 si ad nidum is coepit accedere, procurrit ad pedes eius feta, praegravem aut delumbem sese simulans, subitoque in procursu aut brevi aliquo volatu cadit, ut fracta ala aut pedibus, at procurrit iterum iamiam prehensurum effugiens spemque frustrans, donec in diversum abducat a nidis. eadem in pavore, libera ac materna vacans cura, in sulco resupina glaeba se terrae pedibus adprehensa operit. perdicum vita et ad sedecim 

 annos durare existimatur. 
 partridge 
 si ad nidum is coepit accedere, procurrit ad pedes eius feta, praegravem aut delumbem sese simulans, subitoque in procursu aut brevi aliquo volatu cadit, ut fracta ala aut pedibus, at procurrit iterum iamiam prehensurum effugiens spemque frustrans, donec in diversum abducat a nidis. eadem in pavore, libera ac materna vacans cura, in sulco resupina glaeba se terrae pedibus adprehensa operit. perdicum vita et ad sedecim 

 annos durare existimatur. 
 nest 10 
 104 
 Ab iis columbarum maxime spectantur simili ratione mores. inest pudicitia illis plurima et neutri nota adulteria. coniugi fidem non violant communemque servant domum. nisi caelebs a

# CQ2 - Quelles anecdotes mettent en relation un homme et un animal ?

Cette question exprime le besoin du chercheur à cibler les passages dans le texte où l'auteur mentionne une relation entre un animal et l'homme. Cette relation est décrite au travers d'une anecdote. Les relations comme la chasse et autres ne sont pas attendu. Uniquement les situations individuelles.

   <u>__Reformulation__</u> : La reformulation de la requête SPARQL est très complexe. Pour étudier les possibilités offertes, la requête sera décomposé en plusieurs requêtes portant sur différent concept.

   <u>__Restriction__</u> : Pour restreindre les résultats et apporté de la cohérence entre les deux concepts, l'apparition des concepts mentionnés plus haut doivent faire partie du même paragraphe.

   <u>__Note__</u> : Une anecdote est un concept très difficile à caractériser. La définition proposée est la suivante: une anecdote est un évènemment unique, daté mettant en relation un antroponyme et un animal dans un lieu précisé. Toutes ces informations peuvent ne pas apparaitre dans le texte. De plus, il est possible que lors de l'annotation, le concept d'anecdote n'ait pas été renseigné.


## CQ2.1 - Recherche restreinte : Animal et Anecdote
Cette première approche permet de voir le nombre d'anecdote présent dans les textes ayant un mention d'animal à proximité (même paragraphe).

In [12]:
SELECT DISTINCT ?paragraph ?chapter_id ?paragraph_id ?name_animal ?mention_animal ?mention_anecdote WHERE {
    
?annotation1 a oa:Annotation;
    oa:hasBody ?animal;
    oa:hasTarget [
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_animal; oa:start ?start; oa:end ?end
        ]
    ].

?annotation2 a oa:Annotation;
    oa:hasBody ?anecdote;
    oa:hasTarget [
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_anecdote; oa:start ?start; oa:end ?end
        ]
    ].

?animal a skos:Concept;
    skos:prefLabel ?name_animal.
    
?animal_collection a skos:Collection;
    skos:prefLabel "Ancient class"@en;
    skos:member ?animal.

?anecdote skos:prefLabel "anecdote"@en.

?paragraph schema:text ?text;
           schema:identifier ?paragraph_id;
           schema:isPartOf ?chapter.

?chapter schema:identifier ?chapter_id.

FILTER (lang(?name_animal) = "en").
}
ORDER BY ?paragraph

paragraph 
 chapter_id 
 paragraph_id 
 name_animal 
 mention_animal 
 mention_anecdote http://www.zoomathia.com/Pline/10/121 
 10 
 121 
 raven 
 Reddatur et corvis sua gratia, indignatione quoque populi Romani testata, non solum conscientia. Tiberio principe ex fetu supra Castorum aedem genito pullus in adpositam sutrinam devolavit, etiam religione commendatus officinae domino. is mature sermoni adsuefactus, omnibus matutinis evolans in rostra in forum versus, Tiberium, dein Germanicum et Drusum Caesares nominatim, mox transeuntem populum Romanum salutabat, postea ad tabernam remeans, plurium annorum adsiduo officio mirus. 
 Reddatur et corvis sua gratia, indignatione quoque populi Romani testata, non solum conscientia. Tiberio principe ex fetu supra Castorum aedem genito pullus in adpositam sutrinam devolavit, etiam religione commendatus officinae domino. is mature sermoni adsuefactus, omnibus matutinis evolans in rostra in forum versus, Tiberium, dein Germanicum et Drusum Caesares nominatim, mox transeuntem populum Romanum salutabat, postea ad tabernam remeans, plurium annorum adsiduo officio mirus. http://www.zoomathia.com/Pline/10/122 
 10 
 122 
 raven 
 hunc sive aemulatione vicinitatis manceps proximae sutrinae sive iracundia subita, ut voluit videri, excrementis eius posita calceis macula, exanimavit, tanta plebei consternatione, ut primo pulsus ex ea regione, mox interemptus sit funusque aliti innumeris celebratum exequiis, constratum lectum super Aethiopum duorum umeros, praecedente tibicine et coronis omnium generum ad rogum usque, qui constructus dextra viae Appiae ad secundum lapidem in campo Rediculi appellato fuit. 
 hunc sive aemulatione vicinitatis manceps proximae sutrinae sive iracundia subita, ut voluit videri, excrementis eius posita calceis macula, exanimavit, tanta plebei consternatione, ut primo pulsus ex ea regione, mox interemptus sit funusque aliti innumeris celebratum exequiis, constratum lectum super Aethiopum duorum umeros, praecedente tibicine et coronis omnium generum ad rogum usque, qui constructus dextra viae Appiae ad secundum lapidem in campo Rediculi appellato fuit. http://www.zoomathia.com/Pline/10/123 
 10 
 123 
 raven 
 adeo satis iusta causa populo Romano visa est exequiarum ingenium avis aut supplicii de cive Romano in ea urbe, in qua multorum principum nemo deduxerat funus, Scipionis vero Aemiliani post Carthaginem Numantiamque deletas ab eo nemo vindicaverat mortem. 
 adeo satis iusta causa populo Romano visa est exequiarum ingenium avis aut supplicii de cive Romano in ea urbe, in qua multorum principum nemo deduxerat funus, Scipionis vero Aemiliani post Carthaginem Numantiamque deletas ab eo nemo vindicaverat mortem. http://www.zoomathia.com/Pline/10/124 
 10 
 124 
 raven 
 hoc gestum M. Servilio C. Cestio cos. a. d. V kal. Apriles. nunc quoque erat in urbe Roma, haec prodente me, equitis Romani cornix e Baetica primum colore mira admodum nigro, dein plura contexta verba exprimens et alia atque alia crebro addiscens. nec non et recens fama Crateri Monocerotis cognomine in Erizena regione Asiae corvorum opera venantis eo quod devehebat in silvas eos insidentes corniculo umerisque; illi vestigabant agebantque eo perducta consuetudine, ut exeuntem sic comitarentur et feri. 
 hoc gestum M. Servilio C. Cestio cos. a. d. V kal. Apriles. nunc quoque erat in urbe Roma, haec prodente me, equitis Romani cornix e Baetica primum colore mira admodum nigro, dein plura contexta verba exprimens et alia atque alia crebro addiscens. nec non et recens fama Crateri Monocerotis cognomine in Erizena regione Asiae corvorum opera venantis eo quod devehebat in silvas eos insidentes corniculo umerisque; illi vestigabant agebantque eo perducta consuetudine, ut exeuntem sic comitarentur et feri. http://www.zoomathia.com/Pline/10/124 
 10 
 124 
 crow 
 hoc gestum M. Servilio C. Cestio cos. a. d. V kal. Apriles. nunc quoque erat in urbe Roma, haec prodente me, equitis Romani cornix e Baetica primum colore m

Evalutation des résultats, traduction remacle.

- livre 10 paragraphe 121 à 124:

[121]"Rendons aussi justice aux corbeaux, dont le mérite a été attesté non seulement par le sentiment du peuple romain, mais aussi par son indignation. Sous le règne de Tibère, un petit, né dans un nid placé sur le temple des Dioscures, tomba dans une boutique de cordonnier située vis-à-vis : la religion même le recommandait au maître de la boutique. **L'oiseau, habitué de bonne heure à parler, s'envolait tous les matins sur la tribune, et, tourné vers le forum, il saluait nominativement Tibère, puis les Césars Germanicus et Drusus, puis le peuple qui passait sur la place; après, il retournait dans la boutique. Son assiduité fit pendant plusieurs années l'admiration générale.**"

[122] "Un cordonnier voisin le tua, soit par jalousie, soit par un accès soudain de colère, comme il voulut le faire croire, parce que l'oiseau lui avait sali des chaussures par ses excréments. La multitude en conçut tant de fureur, que d'abord elle chassa de ce quartier, puis tua le coupable. Une foule innombrable assista aux funérailles solennelles de l'oiseau; le lit funéraire fut porté sur les épaules de deux Éthiopiens précédés d'un joueur de flûte, avec des couronnes de toute espèce, jusqu'au bûcher, qui était élevé à la droite de la voie Appienne, à deux milles de Rome, dans le champ appela Rediculus."

[123] " Ainsi le talent d'un oiseau parut au peuple romain une juste cause de faire des funérailles solennelles, ou de punir de mort un citoyen, dans une ville où aucun cortège n'avait suivi le convoi de tant d'hommes remarquables, et où personne n'avait vengé la mort de Scipio Émilien, destructeur de Carthage et de Numance. **Ce fait se passa sous te consulat de M. Servilius et de C. Cestius, le 5 avant les kalendes d'avril (28 mars)**. Aujourd'hui même, au moment où j'écris, il y a dans Rome une corneille qui appartient à un chevalier romain : elle vient de le Bétique. Remarquable par sa couleur absolument noire, elle prononce en outre des phrases entières, et chaque jour elle en apprend de nouvelles."

[124] "Récemment on a parlé de Craterus, surnommé Monoceros, qui, dans l'Erizène, contrée d'Asie, chassait à l'aide de corbeaux. Il les portait dans les forêts, perchés sur les aigrettes de son casque et sur ses épaules; les corbeaux cherchaient le gibier, et le faisaient lever; l'habitude en était tellement prise, que dans ses parties de chasse il était accompagné même par les corbeaux sauvages. Des auteurs ont cru digne de mémoire le fait suivant : Un corbeau altéré fut aperçu jetant des pierres dans une urne funéraire, où de l'eau de pluies était amassée; l'oiseau n'y pouvait pas atteindre, et il craignait de descendre au fond du vase. Par cet amas de pierres il fit monter assez l'eau pour boire."

Anecdote datée, géolocalisée, avec un anthroponyme. La relation est sous-jacente (amitié). Absence de précision dans l'annotation, seulement l'animal et l'anecdote. 2 anecdotes présente ici.

- livre 9 paragraphe 67:

"Asinius Celer, personnage consulaire, prodigue pour ce poisson, en acheta un, sous le règne de Caligula, au prix de 8,000 sesterces (1 168 fr.). Cette prodigalité porte la pensée sur ceux qui, dans leurs doléances sur le luxe, se plaignaient qu'un cuisinier coûtât plus cher qu'un cheval; mais aujourd'hui un poisson coûte le prix d'un cuisinier, un cuisinier le prix d'un triomphe ; et maintenant il n'y a guère d'homme plus estimé que celui qui sait le mieux ruiner son maître. (XVIII) Licinius Mucianus a rapporté qu'un rouget de 80 livres avait été pris dans la mer Rouge. Combien nos gastronomes l'auraient-ils payé s'il avait été péché dans la mer qui baigne nos faubourgs?"

Une anecdote sur le prix d'un rouget mais pas une relation homme - animal.



## CQ2.2 - Nouveau filtre (2.1) : ajout de la relation
La recherche doit prendre en compte la mention d'une relation (utilisation des concepts enfant de "special relationship").

<u>**Sortie**</u> : pas de résultat satisfaisant le filtre. Les paragraphes ayant mentionnant un animal, une anecdote et une relation spécial ne semble pas présent.

In [6]:
SELECT DISTINCT ?chapter_id ?paragraph_id ?name_animal ?mention_animal ?mention_anecdote WHERE {
    
?annotation1 a oa:Annotation;
    oa:hasBody ?animal;
    oa:hasTarget [
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_animal; oa:start ?start; oa:end ?end
        ]
    ].

?annotation2 a oa:Annotation;
    oa:hasBody ?anecdote;
    oa:hasTarget [
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_anecdote; oa:start ?start; oa:end ?end
        ]
    ].

?annotation3 a oa:Annotation;
    oa:hasBody ?relationship;
    oa:hasTarget [
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_relationship; oa:start ?start; oa:end ?end
        ]
    ].

?animal a skos:Concept;
    skos:prefLabel ?name_animal.
    
?animal_collection a skos:Collection;
    skos:prefLabel "Ancient class"@en;
    skos:member ?animal.

?anecdote skos:prefLabel "anecdote"@en.
    
?relationship skos:prefLabel ?name_relationship;
    skos:broader+ ?generic_relationship.
?generic_relationship skos:prefLabel "special relationship"@en.
    
?paragraph schema:text ?text;
           schema:identifier ?paragraph_id;
           schema:isPartOf ?chapter.

?chapter schema:identifier ?chapter_id.

FILTER (lang(?name_animal) = "en").
}
ORDER BY ?paragraph

chapter_id 
 paragraph_id 
 name_animal 
 mention_animal 
 mention_anecdote Total: 0, Shown: 0

Pas de résultat alors qu'il y a bien des anecdotes mettant en relation un homme et un animal. (CF: B10-P121)

### Analyse des concepts annotés dans le paragraphe 121 du livre 10

Pour comprendre pourquoi nous n'avons pas de résultat, il faut regarder les concepts qui ont été annotés.
Comme le montre le tableau réponse ci-dessous, la notion de relation spécial ne figure pas dans la liste des concepts du paragraphe annotés. Il faut donc compléter l'annotation.

In [14]:
SELECT DISTINCT ?concept ?concept_label WHERE {
    
?annotation1 a oa:Annotation;
    oa:hasBody ?concept;
    oa:hasTarget [
        oa:hasSource <http://www.zoomathia.com/Pline/10/121>;
        oa:hasSelector [
            oa:exact ?mention_animal; oa:start ?start; oa:end ?end
        ]
    ].

?concept skos:prefLabel ?concept_label.
filter (lang(?concept_label) = "en")

}

concept 
 concept_label https://opentheso.huma-num.fr/?idc=105291&idt=th310 
 baby https://opentheso.huma-num.fr/?idc=105320&idt=th310 
 anecdote https://opentheso.huma-num.fr/?idc=105379&idt=th310 
 locomotion form https://opentheso.huma-num.fr/?idc=105383&idt=th310 
 flying https://opentheso.huma-num.fr/?idc=105432&idt=th310 
 linguistic capacity https://opentheso.huma-num.fr/?idc=105806&idt=th310 
 historical character https://opentheso.huma-num.fr/?idc=105822&idt=th310 
 Germanicus Caesar https://opentheso.huma-num.fr/?idc=105869&idt=th310 
 imitation of human voice https://opentheso.huma-num.fr/?idc=108569&idt=th310 
 Tiberius https://opentheso.huma-num.fr/?idc=108571&idt=th310 
 Drusus Julius Caesar https://opentheso.huma-num.fr/?idc=pcrtYhYMiLwDUr&idt=th310 
 temple https://opentheso.huma-num.fr/?idc=pcrtfZTkOokDSS&idt=th310 
 Dioscuri https://opentheso.huma-num.fr/?idc=pcrtkERiWybjc4&idt=th310 
 Romans https://opentheso.huma-num.fr/?idc=pcrtva7NUywIgB&idt=th310 
 raven https://opentheso.huma-num.fr/?idg=MT_5&idt=th310 
 People Total: 15, Shown: 15

<hr>

# QC3 - Quels sont les oiseaux qui sont consommés (gastronomie)

Cette question exprime le besoin des chercheurs à identifier les passages dans le texte où il est question des espèces d'oiseaux qui sont utilisées dans la gastronomie.

<u>**Reformulation :**</u> cette question s'exprime très facilement en SPARQL et revient à chercher : (1) une annotation ayant un concept animal  descendant du concept oiseau et (2) un concept descendant du concept "animaux dans la nourriture humaine".

<u>**Note :**</u> Le concept générique "Oiseau" ne couvrent pas l'ensemble des concepts d'oiseau du thésaurus. D'autre hierarchie existe pour qualifier certaines espèces d'oiseau (oiseau palmipède, oiseau avec griffe etc). Par conséquent, il faut prendre en compte tous ces concepts dans la recherche. Une solution possible pour "corriger" ce problème est de reconsidérer la place du concept "OISEAU" dans la hiérarchie sachant que les autres sont, en soit, généralisable comme étant des oiseaux.

<u>**Remarque :**</u> L'absence de résultat entre les deux versions de ZooKG s'explique par l'absence du concept "cooking" ignoré lors de la génération car étant ambiguë (55 occurences). Cette ambiguité a été réglé dans la nouvelle version du thésaurus.

In [15]:
SELECT DISTINCT ?paragraph ?name_animal ?mention_animal ?name_conso ?mention_conso WHERE {
  ?annotation1 oa:hasBody ?animal; oa:hasTarget [
    oa:hasSource ?paragraph;
    oa:hasSelector [
        oa:exact ?mention_animal
        ]
    ].

  ?animal skos:prefLabel ?name_animal; skos:broader+ ?animal_generic.
    
  ?animal_generic skos:prefLabel ?name_animal_generic.

  ?annotation2 oa:hasBody ?conso;
        oa:hasTarget  [oa:hasSource ?paragraph;
      oa:hasSelector [oa:exact ?mention_conso]].

  ?conso skos:prefLabel ?name_conso; skos:broader+ ?conso_generique.
  ?conso_generique skos:prefLabel "animal in human nourishing"@en.
FILTER (?name_animal_generic in ("BIRD"@en,"BIRD WITH CLAW"@en,
                                 "BIRD WITH TOES"@en,"WATERBIRD"@en,"NIGHT BIRD"@en,"MIGRATORY BIRD"@en, "PALMIPED BIRD"@en)).
}
ORDER BY ?paragraph

paragraph 
 name_animal 
 mention_animal 
 name_conso 
 mention_conso Total: 0, Shown: 0

# CQ4 - Quels sont les usages thérapeutiques des parties d'animaux ?

Cette question est dérivé de la question originelle "Quels sont les remèdes (thérapeutiques) qui incluent une langue animale (ou un morceau de langue)?" qui ne pouvait être exprimer en SPARQL à cause de l'absence d'annotation relatif aux remèdes et l'absence du concept même dans le thesaurus. Ce concept sera ajouté lors de l'annotation des livres 28 à 30 qui traitent des remèdes et utilisation médicale des animaux.

<u>**Reformulation :**</u> Cette question s'exprime en SPARQL en recherchant : (1) un concept animal et (2) un concept descendant du concept "utilisation médical de l'animal".

<u>**Note :**</u> La partie animal n'est pas précisé.

In [17]:
SELECT DISTINCT ?paragraph ?name_animal ?mention_animal ?name_use ?mention_use WHERE {
?annotation1 oa:hasBody ?animal; 
    oa:hasTarget [
        oa:hasSource ?paragraph; 
        oa:hasSelector [
            oa:exact ?mention_animal;
            oa:start ?start_mention_animal;
            oa:end ?end_mention_animal;
            ]
        ].

?annotation2 oa:hasBody ?use;
    oa:hasTarget [
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_use;
            oa:start ?start_mention_use;
            oa:end ?end_mention_use;
        ]
    ].

?animal skos:prefLabel ?name_animal.
    
?animal_collection skos:prefLabel "Ancient class"@en;
    skos:member ?animal.

?use skos:prefLabel ?name_use;
    skos:broader+ ?use_generique.

?use_generique skos:prefLabel "medical use of animal"@en.
  
FILTER (lang(?name_animal) = "en").
FILTER (lang(?name_use) = "en")
}
ORDER BY ?paragraph

paragraph 
 name_animal 
 mention_animal 
 name_use 
 mention_use http://www.zoomathia.com/Pline/10/55 
 goose 
 Aliud repperit Syriae pars quae Commagene vocatur, adipem eorum in vase aereo cum cinnamo nive multa obrutum ac rigore gelido maceratum ad usum praeclari medicaminis, quod ab gente dicitur Commagenum. 
 medical use of part of animal 
 Aliud repperit Syriae pars quae Commagene vocatur, adipem eorum in vase aereo cum cinnamo nive multa obrutum ac rigore gelido maceratum ad usum praeclari medicaminis, quod ab gente dicitur Commagenum. http://www.zoomathia.com/Pline/11/118 
 gnat 
 spirant. si adcrevere, faciunt chrysallidem. ficarios culices caprificus generat, cantharidas vermiculi ficorum et piri et

 peuces et cynacanthae et rosae. venenum hoc remedia secum habet: alae medentur, quibus demptis letale est. rursus alia genera culicum acescens natura gignit, quippe cum et in nive inveniantur vetustiore vermiculi rutili_nam et ipsa nix vetustate rufescit_ hirti pilis, in Media quidem candidi, grandiores torpentesque. 
 medical use of part of animal 
 spirant. si adcrevere, faciunt chrysallidem. ficarios culices caprificus generat, cantharidas vermiculi ficorum et piri et

 peuces et cynacanthae et rosae. venenum hoc remedia secum habet: alae medentur, quibus demptis letale est. rursus alia genera culicum acescens natura gignit, quippe cum et in nive inveniantur vetustiore vermiculi rutili_nam et ipsa nix vetustate rufescit_ hirti pilis, in Media quidem candidi, grandiores torpentesque. http://www.zoomathia.com/Pline/11/118 
 VERMIS 
 spirant. si adcrevere, faciunt chrysallidem. ficarios culices caprificus generat, cantharidas vermiculi ficorum et piri et

 peuces et cynacanthae et rosae. venenum hoc remedia secum habet: alae medentur, quibus demptis letale est. rursus alia genera culicum acescens natura gignit, quippe cum et in nive inveniantur vetustiore vermiculi rutili_nam et ipsa nix vetustate rufescit_ hirti pilis, in Media quidem candidi, grandiores torpentesque. 
 medical use of part of animal 
 spirant. si adcrevere, faciunt chrysallidem. ficarios culices caprificus generat, cantharidas vermiculi ficorum et piri et

 peuces et cynacanthae et rosae. venenum hoc remedia secum habet: alae medentur, quibus demptis letale est. rursus alia genera culicum acescens natura gignit, quippe cum et in nive inveniantur vetustiore vermiculi rutili_nam et ipsa nix vetustate rufescit_ hirti pilis, in Media quidem candidi, grandiores torpentesque. http://www.zoomathia.com/Pline/11/118 
 caterpillar with robe 
 spirant. si adcrevere, faciunt chrysallidem. ficarios culices caprificus generat, cantharidas vermiculi ficorum et piri et

 peuces et cynacanthae et rosae. venenum hoc remedia secum habet: alae medentur, quibus demptis letale est. rursus alia genera culicum acescens natura gignit, quippe cum et in nive inveniantur vetustiore vermiculi rutili_nam et ipsa nix vetustate rufescit_ hirti pilis, in Media quidem candidi, grandiores torpentesque. 
 medical use of part of animal 
 spirant. si adcrevere, faciunt chrysallidem. ficarios culices caprificus generat, cantharidas vermiculi ficorum et piri et

 peuces et cynacanthae et rosae. venenum hoc remedia secum habet: alae medentur, quibus demptis letale est. rursus alia genera culicum acescens natura gignit, quippe cum et in nive inveniantur vetustiore vermiculi rutili_nam et ipsa nix vetustate rufescit_ hirti pilis, in Media quidem candidi, grandiores torpentesque. http://www.zoomathia.com/Pline/11/118 
 fig gnat 
 spirant. si adcrevere, faciunt chrysallidem. ficarios culices caprificus generat, cantharidas vermiculi ficorum et piri et

 peuces et cynacanthae et rosae. venenum hoc remedia secum habet: alae medentur, quibus demptis letale est. rursus alia genera culicum acescens natura gignit, quippe cum et in nive inveniantur vetustiore vermiculi rutili_nam et ipsa nix vetustate rufescit_ hirti pilis, in Media quidem candidi, grandiores torpentesque. 
 medical use of part of anima

# CQ5 - Quels sont les animaux qui communiquent entre eux ?

Cette requête exprime le besoin des chercheurs à identifier les animaux capables de communiquer au sein de la même espèces.

<u>**Reformulation :**</u> Pour représenter cette question en SPARQL, nous cherchons les paragraphes qui mentionnent : (1) un concept d'animal et (2) un concept de parole ou "capacité linguistique".

<u>**Note :**</u> Ce sont bien des communications **intra**-espèce et non inter-espèce qui intéresse le chercheur dans ce cas-ci.

<u>**Remarques :**</u> Le concept "capacité linguistique" est trop générique et inclu les espèces qui sont capables de reproduire des sonorités humaines. Cela de veut pas dire que ces espèces communiquent.

In [23]:
SELECT DISTINCT ?paragraph ?name_animal ?mention_animal ?name_social ?mention_social WHERE {
    
?annotation1 a oa:Annotation;
    oa:hasBody ?animal;
    oa:hasTarget [
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_animal; oa:start ?start_mention_animal; oa:end ?end_mention_animal;
        ]
    ].

?animal skos:prefLabel ?name_animal.
    
?animal_collection skos:prefLabel ?collection_name;
    skos:member ?animal.

?annotation2 oa:hasBody ?social;
    oa:hasTarget [
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_social;
            oa:start ?start_mention_social;
            oa:end ?end_mention_social;
        ]
    ].
    
?social skos:prefLabel ?name_social.
FILTER (lang(?name_animal) = "en")
FILTER (?collection_name in ("Ancient class"@en, "Zoonyms"@en))
FILTER (?name_social in ("speech"@en, "linguistic capacity"@en))
}
ORDER BY ?paragraph

paragraph 
 name_animal 
 mention_animal 
 name_social 
 mention_social http://www.zoomathia.com/Pline/10/117 
 parrot 
 super omnia humanas voces reddunt, psittaci quidem etiam sermocinantes. India hanc avem mittit, siptacen vocat, viridem toto corpore, torque tantum miniato in cervice distinctam. imperatores salutat et quae accipit verba pronuntiat, in vino praecipue lasciva. capiti eius duritia eadem quae rostro. hoc, cum loqui discit, ferreo verberatur radio; non sentit aliter ictus. cum devolat, rostro se excipit, illi innititur levioremque ita se pedum infirmitati facit. 
 linguistic capacity 
 super omnia humanas voces reddunt, psittaci quidem etiam sermocinantes. India hanc avem mittit, siptacen vocat, viridem toto corpore, torque tantum miniato in cervice distinctam. imperatores salutat et quae accipit verba pronuntiat, in vino praecipue lasciva. capiti eius duritia eadem quae rostro. hoc, cum loqui discit, ferreo verberatur radio; non sentit aliter ictus. cum devolat, rostro se excipit, illi innititur levioremque ita se pedum infirmitati facit. http://www.zoomathia.com/Pline/10/117 
 parrot 
 super omnia humanas voces reddunt, psittaci quidem etiam sermocinantes. India hanc avem mittit, siptacen vocat, viridem toto corpore, torque tantum miniato in cervice distinctam. imperatores salutat et quae accipit verba pronuntiat, in vino praecipue lasciva. capiti eius duritia eadem quae rostro. hoc, cum loqui discit, ferreo verberatur radio; non sentit aliter ictus. cum devolat, rostro se excipit, illi innititur levioremque ita se pedum infirmitati facit. 
 speech 
 super omnia humanas voces reddunt, psittaci quidem etiam sermocinantes. India hanc avem mittit, siptacen vocat, viridem toto corpore, torque tantum miniato in cervice distinctam. imperatores salutat et quae accipit verba pronuntiat, in vino praecipue lasciva. capiti eius duritia eadem quae rostro. hoc, cum loqui discit, ferreo verberatur radio; non sentit aliter ictus. cum devolat, rostro se excipit, illi innititur levioremque ita se pedum infirmitati facit. http://www.zoomathia.com/Pline/10/117 
 BIRD 
 super omnia humanas voces reddunt, psittaci quidem etiam sermocinantes. India hanc avem mittit, siptacen vocat, viridem toto corpore, torque tantum miniato in cervice distinctam. imperatores salutat et quae accipit verba pronuntiat, in vino praecipue lasciva. capiti eius duritia eadem quae rostro. hoc, cum loqui discit, ferreo verberatur radio; non sentit aliter ictus. cum devolat, rostro se excipit, illi innititur levioremque ita se pedum infirmitati facit. 
 linguistic capacity 
 super omnia humanas voces reddunt, psittaci quidem etiam sermocinantes. India hanc avem mittit, siptacen vocat, viridem toto corpore, torque tantum miniato in cervice distinctam. imperatores salutat et quae accipit verba pronuntiat, in vino praecipue lasciva. capiti eius duritia eadem quae rostro. hoc, cum loqui discit, ferreo verberatur radio; non sentit aliter ictus. cum devolat, rostro se excipit, illi innititur levioremque ita se pedum infirmitati facit. http://www.zoomathia.com/Pline/10/117 
 BIRD 
 super omnia humanas voces reddunt, psittaci quidem etiam sermocinantes. India hanc avem mittit, siptacen vocat, viridem toto corpore, torque tantum miniato in cervice distinctam. imperatores salutat et quae accipit verba pronuntiat, in vino praecipue lasciva. capiti eius duritia eadem quae rostro. hoc, cum loqui discit, ferreo verberatur radio; non sentit aliter ictus. cum devolat, rostro se excipit, illi innititur levioremque ita se pedum infirmitati facit. 
 speech 
 super omnia humanas voces reddunt, psittaci quidem etiam sermocinantes. India hanc avem mittit, siptacen vocat, viridem toto corpore, torque tantum miniato in cervice distinctam. imperatores salutat et quae accipit verba pronuntiat, in vino praecipue lasciva. capiti eius duritia eadem quae rostro. hoc, cum loqui discit, ferreo verberatur radio; non sentit aliter ictus. cum devolat, rostro se excipit, illi innititur levior

# CQ 6 - Quelles sont les données transmises sur le temps de gestation des animaux ?

<u>**Reformulation :**</u> Pour représenter cette question en SPARQL, nous cherchons les paragraphes qui mentionnent : (1) un concept d'animal et (2) le concept "length of pregnancy".

In [25]:
SELECT DISTINCT ?paragraph ?name_animal ?mention_animal ?mention_pregnancy WHERE {
?annotation1 oa:hasBody ?animal;
    oa:hasTarget  [
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_animal;
            oa:start ?start_mention_animal;
            oa:end ?end_mention_animal;
        ]
    ].
    
?animal skos:prefLabel ?name_animal.
    
?animal_collection skos:prefLabel ?collection_name;
    skos:member ?animal.

?annotation2 oa:hasBody ?pregnancy;
    oa:hasTarget [
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_pregnancy;
            oa:start ?start_mention_pregnancy;
            oa:end ?end_mention_pregnancy;
        ]
    ].

?pregnancy skos:prefLabel "length of pregnancy"@en.

FILTER (lang(?name_animal) = "en").
FILTER (?collection_name in ("Ancient class"@en, "Zoonyms"@en))
}
ORDER BY ?paragraph

paragraph 
 name_animal 
 mention_animal 
 mention_pregnancy http://www.zoomathia.com/Pline/10/147 
 swallow1 
 columbae deciens anno pariunt, quaedam et undeciens, in Aegypto vero etiam brumali mense. hirundines et merulae et palumbi et turtures bis anno pariunt, ceterae aves fere semel. turdi, in cacuminibus arborum luto nidificantes paene contextim, in secessu generant. a coitu X diebus ova maturescunt in utero, vexatis autem gallinae et columbae pinna evulsa aliave simili iniuria diutius. 
 columbae deciens anno pariunt, quaedam et undeciens, in Aegypto vero etiam brumali mense. hirundines et merulae et palumbi et turtures bis anno pariunt, ceterae aves fere semel. turdi, in cacuminibus arborum luto nidificantes paene contextim, in secessu generant. a coitu X diebus ova maturescunt in utero, vexatis autem gallinae et columbae pinna evulsa aliave simili iniuria diutius. http://www.zoomathia.com/Pline/10/147 
 ringdove 
 columbae deciens anno pariunt, quaedam et undeciens, in Aegypto vero etiam brumali mense. hirundines et merulae et palumbi et turtures bis anno pariunt, ceterae aves fere semel. turdi, in cacuminibus arborum luto nidificantes paene contextim, in secessu generant. a coitu X diebus ova maturescunt in utero, vexatis autem gallinae et columbae pinna evulsa aliave simili iniuria diutius. 
 columbae deciens anno pariunt, quaedam et undeciens, in Aegypto vero etiam brumali mense. hirundines et merulae et palumbi et turtures bis anno pariunt, ceterae aves fere semel. turdi, in cacuminibus arborum luto nidificantes paene contextim, in secessu generant. a coitu X diebus ova maturescunt in utero, vexatis autem gallinae et columbae pinna evulsa aliave simili iniuria diutius. http://www.zoomathia.com/Pline/10/147 
 blackbird 
 columbae deciens anno pariunt, quaedam et undeciens, in Aegypto vero etiam brumali mense. hirundines et merulae et palumbi et turtures bis anno pariunt, ceterae aves fere semel. turdi, in cacuminibus arborum luto nidificantes paene contextim, in secessu generant. a coitu X diebus ova maturescunt in utero, vexatis autem gallinae et columbae pinna evulsa aliave simili iniuria diutius. 
 columbae deciens anno pariunt, quaedam et undeciens, in Aegypto vero etiam brumali mense. hirundines et merulae et palumbi et turtures bis anno pariunt, ceterae aves fere semel. turdi, in cacuminibus arborum luto nidificantes paene contextim, in secessu generant. a coitu X diebus ova maturescunt in utero, vexatis autem gallinae et columbae pinna evulsa aliave simili iniuria diutius. http://www.zoomathia.com/Pline/10/147 
 turtledove 
 columbae deciens anno pariunt, quaedam et undeciens, in Aegypto vero etiam brumali mense. hirundines et merulae et palumbi et turtures bis anno pariunt, ceterae aves fere semel. turdi, in cacuminibus arborum luto nidificantes paene contextim, in secessu generant. a coitu X diebus ova maturescunt in utero, vexatis autem gallinae et columbae pinna evulsa aliave simili iniuria diutius. 
 columbae deciens anno pariunt, quaedam et undeciens, in Aegypto vero etiam brumali mense. hirundines et merulae et palumbi et turtures bis anno pariunt, ceterae aves fere semel. turdi, in cacuminibus arborum luto nidificantes paene contextim, in secessu generant. a coitu X diebus ova maturescunt in utero, vexatis autem gallinae et columbae pinna evulsa aliave simili iniuria diutius. http://www.zoomathia.com/Pline/10/147 
 thrush 
 columbae deciens anno pariunt, quaedam et undeciens, in Aegypto vero etiam brumali mense. hirundines et merulae et palumbi et turtures bis anno pariunt, ceterae aves fere semel. turdi, in cacuminibus arborum luto nidificantes paene contextim, in secessu generant. a coitu X diebus ova maturescunt in utero, vexatis autem gallinae et columbae pinna evulsa aliave simili iniuria diutius. 
 columbae deciens anno pariunt, quaedam et undeciens, in Aegypto vero etiam brumali mense. hirundines et merulae et palumbi et turtures bis anno pariunt, ceterae aves fere semel. turdi, in cacuminibus

# CQ7 - Sur le rythme alimentaire des animaux : quels sont les animaux capables de jeûner, quelles sont les informations sur les rythmes de repas (fréquence)?

<u>**Reformulation :**</u> Pour représenter cette question en SPARQL, nous cherchons les paragraphes qui mentionnent : (1) un concept d'animal et (2) le concept de jeûne (fasting).

<u>**Sortie :**</u> Pour chaque résultat, nous affichons le concept animal, le paragraphe et la mention dans le paragraphe.

<u>**Analyse :**</u> Aucune mention du concept "fasting" n'est présente dans les annotations. Par conséquent, cette requête n'a pas de résultat. Il se trouve que le concept "special diet" a été utilisé dans les annotations manuelles mais n'a pas de correspondance dans le thésaurus TheZoo. Il est peut-être nécessaire de modifier cette annotation.

In [27]:
SELECT DISTINCT ?paragraph ?name_animal ?mention_animal ?mention_fasting WHERE {
?annotation1 oa:hasBody ?animal;
    oa:hasTarget [ 
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_animal;
            oa:start ?start_mention_animal;
            oa:end ?end_mention_animal;
        ]
    ].

?animal skos:prefLabel ?name_animal.    
?animal_collection skos:prefLabel ?collection_name;
       skos:member ?animal.

?annotation2 oa:hasBody ?fasting;
    oa:hasTarget [ 
        oa:hasSource ?paragraph;
        oa:hasSelector [
            oa:exact ?mention_fasting;
            oa:start ?start_mention_fasting;
            oa:end ?end_mention_fasting;
        ]
    ].

?fasting skos:prefLabel "fasting"@en.

FILTER (lang(?name_animal) = "en").
FILTER (?collection_name in ("Ancient class"@en, "Zoonyms"@en))
}
ORDER BY ?paragraph

paragraph 
 name_animal 
 mention_animal 
 mention_fasting Total: 0, Shown: 0